core-bundle/src/Resources/contao/library/Contao/Model/Registry.php

```diff
@@ -75,6 +75,18 @@ public static function getInstance()
 		return static::$objInstance;
 	}
 
+	/**
+	 * Clear the registry and dump all tracking information.
+	 *
+	 * @internal This should only be called by @Contao\CoreBundle\Framework\ContaoFramework during the kernel.reset event.
+	 */
+	public function reset()
+	{
+		$this->arrRegistry = array();
+		$this->arrAliases = array();
+		$this->arrIdentities = array();
+	}
```

Comment:
```suggestion
	public static function reset()
	{
		static::$objInstance = new static();
	}
```

Reply:
This imo looks nicer. Is there a benefit of recreating the object over resetting it's properties other than convenience/style?

Reply:
I don’t think so, no.